litentry/primitives/src/validation_data.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020-2022 Litentry Technologies GmbH.
+// This file is part of Litentry.
+//
+// Litentry is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Litentry is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
+
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+
+use codec::{Decode, Encode, MaxEncodedLen};
+use scale_info::TypeInfo;
+use sp_runtime::{traits::ConstU32, BoundedVec};
+// use sp_runtime::{traits::ConstU32, BoundedVec, MultiSignature};
+
+pub type MaxStringLength = ConstU32<64>;
+pub type ValidationString = BoundedVec<u8, MaxStringLength>;
+
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct TwitterValidationData {
+	pub tweet_id: ValidationString,
+}
+
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct DiscordValidationData {
+	pub channel_id: ValidationString,
+	pub message_id: ValidationString,
+	pub guild_id: ValidationString,
+}
+
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct Web3CommonValidationData {
+	pub message: ValidationString, // or String if under std
+	                               // pub signature: MultiSignature,
```

Comment:
Why not have the signature field? 😂  Ideally we could define the concrete type: MultiSignature for substrate, and `[u8; 65]` for eth-signature I assume

Reply:
Can these multiSig and [u8; 65] be bonded to the specific type? Otherwise it might create confusion.

Reply:
> Why not have the signature field? 😂 Ideally we could define the concrete type: MultiSignature for substrate, and `[u8; 65]` for eth-signature I assume

`MultiSignature` can't be serialize and deserialize

Reply:
But we are only use it off-chain anyway?

Reply:
Sorry, I forgot I had added the feature `std` 😂.  Now, it should be ok